src/views/TradingCompetition/index.tsx

```diff
@@ -115,10 +135,49 @@ const TradingCompetition = () => {
         canClaimNFT: false,
       })
     }
-
-    fetchCompetitionInfo()
+    fetchCompetitionInfoContract()
   }, [account, registrationSuccessful, claimSuccessful, tradingCompetitionContract])
 
+  useEffect(() => {
+    const fetchUserTradingStats = async () => {
+      const res = await fetch(`${profileApiUrl}/api/users/${account}`)
+      const data = await res.json()
+      setUserLeaderboardInformation(data.leaderboard)
+    }
+
+    const fetchGlobalLeaderboardStats = async () => {
+      const res = await fetch(`${profileApiUrl}/api/leaderboard/global`)
+      const data = await res.json()
+      setGlobalLeaderboardInformation(data)
+    }
+
+    const fetchTeamsLeaderboardStats = async (teamId, callBack) => {
+      try {
+        const res = await fetch(`${profileApiUrl}/api/leaderboard/team/${teamId}`)
+        const data = await res.json()
+        callBack(data)
+      } catch (e) {
+        console.error(e)
+      }
+    }
```

Comment:
I think these should go in a helpers file. And please type your functions.

Reply:
Added types & separated into useEffects depending on whether they ought to always be fetched, vs only fetching when a user has registered. DMed you about improving these post-launch